src/H5Oint.c

```diff
@@ -1594,37 +1594,39 @@ H5O_obj_type(const H5O_loc_t *loc, H5O_type_t *obj_type)
 /*-------------------------------------------------------------------------
  * Function:    H5O__obj_type_real
  *
- * Purpose:    Returns the type of object pointed to by `oh'.
+ * Purpose:     On success, returns the type of object pointed to by `oh' or
+ *              NULL in *obj_type.  *obj_type not defined on failure.
  *
- * Return:    Success:    Non-negative
- *        Failure:    Negative
+ * Return:      Success:    Non-negative
+ *              Failure:    Negative
  *
  *-------------------------------------------------------------------------
  */
 static herr_t
 H5O__obj_type_real(const H5O_t *oh, H5O_type_t *obj_type)
 {
-    const H5O_obj_class_t *obj_class; /* Class of object for header */
+    const H5O_obj_class_t *obj_class = NULL; /* Class of object for header */
+    herr_t                 ret_value = SUCCEED;
 
-    FUNC_ENTER_PACKAGE_NOERR
+    FUNC_ENTER_PACKAGE
 
     /* Sanity check */
     assert(oh);
     assert(obj_type);
 
     /* Look up class for object header */
-    if (NULL == (obj_class = H5O__obj_class_real(oh))) {
-        /* Clear error stack from "failed" class lookup */
-        H5E_clear_stack();
+    if (H5O__obj_class_real(oh, &obj_class) < 0)
+        HGOTO_ERROR(H5E_OHDR, H5E_CANTGET, FAIL, "unable to determine object class");
 
-        /* Set type to "unknown" */
-        *obj_type = H5O_TYPE_UNKNOWN;
-    }
-    else
+    if (obj_class)
         /* Set object type */
         *obj_type = obj_class->type;
+    else
```

Comment:
It doesn't seem like it's actually possible for this case to occur -  if `obj_class` is NULL, then that means `H5O__obj_class_real()` encountered an error and the `HGOTO_ERROR` up above was triggered.

Reply:
Agree; I'll update

Reply:
Updated